x/xcvm/keeper/transfer_intent.go

```diff
@@ -0,0 +1,448 @@
+
+	transferIntentKey := types.GetPendingTransferIntentKeyById(intentId)
+	transferIntentValue := k.cdc.MustMarshal(&transferIntent)
+
+	kvStore.Set(transferIntentKey, transferIntentValue)
+}
+
+// GetTransferIntent retrieves a transfer intent from the store
+func (k Keeper) GetTransferIntent(ctx sdk.Context, intentId uint64) (*types.TransferIntent, error) {
+	kvStore := ctx.KVStore(k.storeKey)
+
+	transferIntentKey := types.GetPendingTransferIntentKeyById(intentId)
+	if !kvStore.Has(transferIntentKey) {
+		return nil, types.ErrInvalidIntentId
+	}
+
+	transferIntentBz := kvStore.Get(transferIntentKey)
+	var transferIntent types.TransferIntent
+	if err := k.cdc.Unmarshal(transferIntentBz, &transferIntent); err != nil {
+		return nil, err
+	}
+
+	return &transferIntent, nil
+}
+
+// VerifyEthTransferIntentProof verifies the proof submitted by the solver to claim the bounty for executing the transfer intent
+func (k Keeper) VerifyEthTransferIntentProof(ctx sdk.Context, msg *types.MsgVerifyTransferIntentProof) error {
+	kvStore := ctx.KVStore(k.storeKey)
+
+	// verify that the transfer intent exists
+	transferIntent, err := k.GetTransferIntent(ctx, msg.IntentId)
+	if err != nil {
+		return fmt.Errorf("get transfer intent: %v", err)
+	}
+
+	// verify that the transfer intent request is not timed out
+	currentBlockHeight := ctx.BlockHeight()
+	if currentBlockHeight >= transferIntent.TimeoutHeight {
+		return types.ErrProofSubmittedAfterTimeout
+	}
+
+	// verify that the transaction receipt is valid and included in the execution block header
+	var txReceipt gethtypes.Receipt
+	if err := txReceipt.UnmarshalJSON(msg.TxReceipt); err != nil {
+		return types.ErrInvalidTxReceipt
+	}
+	var blockHeader gethtypes.Header
+	if err := rlp.DecodeBytes(msg.BlockHeader, &blockHeader); err != nil {
+		return fmt.Errorf("decode block header: %v", err)
+	}
+	txReceiptHash, err := getTxReceiptHash(txReceipt)
+	if err != nil {
+		return fmt.Errorf("get tx receipt hash: %v", err)
+	}
+	if err = verifyReceiptProof(blockHeader, txReceiptHash, msg.ReceiptProof); err != nil {
+		return fmt.Errorf("verify receipt proof: %v", err)
+	}
+
+	clientId := transferIntent.ClientId
+	clientState, err := k.getClientState(ctx, clientId)
+	if err != nil {
+		return fmt.Errorf("get client state: %v", err)
+	}
+
+	clientStateBeaconBlockHeader, err := getClientStateBeaconBlockHeader(clientState)
+	if err != nil {
+		return fmt.Errorf("get client state beacon block header: %v", err)
+	}
+
+	// If the light client is ahead of the block where the intent was executed, verify that the provided beacon block headers are valid
+	var intentBeaconBlockHeader *types.BeaconBlockHeader
+	if len(msg.BeaconBlockHeaders) > 0 {
+		// Include the light client state as the last block header in the chain to verify
+		beaconBlockHeaders := append(msg.BeaconBlockHeaders, clientStateBeaconBlockHeader)
+		if err := verifyPreviousBeaconBlockHeaders(beaconBlockHeaders); err != nil {
```

Comment:
Should this func just be called `verifyBeaconBlockHeaders` since the current light client beacon block header is added to the list of headers as well?